File: python3/LinkedList.py
class Node:

    def __init__(self, data):
        self.data = None
        self.next = Node()

class LinkedList:

    def __init__(self):
        self.head = None

    def append(self,data):
        if self.head is None:
            self.head = Node(data)
            return
        current = self.head
        while current.next is not None:
            current = current.next
        current.next = Node(data)
        return current

    def prepend(self,data):
        if self.head is None:
            self.head = Node(data)
            return
        newHead = Node(data)
        newHead.next = self.head
        self.head = newHead
        return

    def delete_with_value(self,data):
        if self.head is None:
            return
        if self.head.data == data:
            self.head = self.head.next
            return
        current = self.head
        while current.head is not None:
            if current.next.data == data:
                current.next = current.next.next
            current = current.next
        return
        

File: java/LinkedList.java
public class Node {
    Node next;
    int data;

    public Node(int data) {
        this.data = data
    }
}

public class LinkedList{
    Node head;

    public void append(int data) {
        if (head == null) {
            head = new Node(data);
            return;
        }
        Node current = head;
        while (current.next != null) {
            current = current.next;
        }
        current.next = new Node(data);
        return;
    }

    public void prepend(int data) {
        if (head == null) {
            head = new Node(data);
            return;
        }
        Node newHead = new Node(data);
        newHead.next = head;
        head = newHead;
        return;
    }

    public void delete_with_value(int data) {
        if (head == null) {
            return;
        }
        if (head.data == data) {
            head = head.next;
            return;
        }
        Node current = head;
        while (current.next != null) {
            if (current.next == data) {
                current.next = current.next.next;
            current = current.next;
            }
        }
    }

}